src/robotide/editor/texteditor.py

```diff
@@ -1056,6 +1057,105 @@ def execute_uncomment(self, event):
+        # If the selection spans more than one line:
+        if ini_line < end_line:   # TODO: do inserts in such a way that they can be undone in 1 undo
+            cells_to_insert = cell_tot_ini - cell_no_beg + 1
+            amount_of_lines = end_line - ini_line
+            new_start = cell_pos_beg
+            #print(f"DEBUG: insert {cells_to_insert} cell(s) before cell={cell_no_beg} on amount of lines={amount_of_lines}")
+            for line in range(ini_line, end_line+1):
+                begthis = self._editor.PositionFromLine(line)
+                endthis = self._editor.PositionFromLine(line+1)
+                cell_pos_beg = self._get_position_of_cell(begthis, endthis, cell_no_beg)
+                self._editor.InsertText(cell_pos_beg, (celltab + cellval) * cells_to_insert)
+            new_end = cell_pos_beg + (len((celltab + cellval).encode('utf-8')) * cells_to_insert)
+        elif start == end:  # On a single row, no selection
+            #print(f"DEBUG: insert 1 cell before cell={cell_no_beg} on line={ini_line}")
+            self._editor.InsertText(cell_pos_beg, celltab + cellval)
+            new_start = cell_pos_beg
+            new_end   = cell_pos_beg + len((celltab + cellval).encode('utf-8'))
+        else:  # On a single row, with selection
+            cells_to_insert = cell_no_end - cell_no_beg + 1
+            #print(f"DEBUG: insert {cells_to_insert} cell(s) before cell={cell_no_beg} on line={ini_line}")
+            # insert at once so undo handles it correct
+            self._editor.InsertText(cell_pos_beg, (celltab + cellval) * cells_to_insert)
+            new_start = cell_pos_beg
+            new_end   = cell_pos_beg + (len((celltab + cellval).encode('utf-8')) * cells_to_insert)
+        # subtract 1 from endpos to circumvent cursor being on end of cell --> otherwise no will be next cell no
+        new_end -= 1
+        # SetSelection and SetCurrentPos + Store_position overrule each other so only use one of them
+        self._editor.SetSelection(new_start, new_end)
+
+    def _get_cell_no(self, begpos, endpos, findpos):
+        # get cell number from range begpos-endpos using findpos 
+        cell_no = 0
+        celltot = self._get_number_of_cells(begpos, endpos)
+        while cell_no < celltot:
+            cell_no += 1
+            cellpos = self._get_position_of_cell(begpos, endpos, cell_no)
+            #print(f"DEBUG loop: celltot={celltot} cell_no={cell_no} cellpos={cellpos} findpos={findpos}")
+            if cellpos > findpos:
+                cell_no -= 1
+                break
+        return cell_no
+
+    def _get_number_of_cells(self, begpos, endpos):
+        # get number of cells in range begpos-endpos
+        # Warning! GetStringSelection does not work properly if there are diacritics in the content above (!) the selected range
+        the_content = self._editor.GetTextRange(begpos, endpos)
+        celltab = ' ' * self._tab_size
```

Comment:
Will this also work when working on files that have a different tab size, but haven't been modified yet?

I ask because I have seen some issues where I had the tab size set to 4, but then when I pulled files from the archive that were 2-space separated, it was interpreted as a single cell.

Reply:
On positioning the current tab size is taken into account as set by _tab_size. Or do you mean something else?
There is a lot of code working with this _tab_size.

On another note: I did not get any mail or so on your review. Should I have gotten some notice?